src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -273,6 +274,17 @@ protected virtual void OnBorderBrushChanged(Brush oldValue, Brush newValue)
 					(s, _) => OnBorderBrushChangedPartial()
 				);
 			}
+			else if (newValue is RadialGradientBrush rgb)
+			{
+				_borderBrushColorChanged.Disposable = rgb.RegisterDisposablePropertyChangedCallback(
+					RadialGradientBrush.FallbackColorProperty,
+					(s, colorArg) => OnBorderBrushChangedPartial()
+				);
+				_borderBrushOpacityChanged.Disposable = rgb.RegisterDisposablePropertyChangedCallback(
+					RadialGradientBrush.OpacityProperty,
+					(s, _) => OnBorderBrushChangedPartial()
+				);
+			}
```

Comment:
We'll need to refactor this one day...  that's definitely not a good pattern and Border knows _too much_ about specific brushes...

Reply:
@carldebilly Agreed. This should somehow be moved to each Brush-specific class. Will look into that as a follow-up

Reply:
@dr1rrb will be pleased :-)

Reply:
I hate this kind of "type enumeration". WinUI is build using inheritance, we have to follow that pattern and take advantage of it when possible by moving that kind of logic in the implementation itself.